clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2302,6 +2302,10 @@ def err_init_list_bad_dest_type : Error<
 def warn_cxx20_compat_aggregate_init_with_ctors : Warning<
   "aggregate initialization of type %0 with user-declared constructors "
   "is incompatible with C++20">, DefaultIgnore, InGroup<CXX20Compat>;
+def warn_cxx20_compat_requires_explicit_init_non_aggregate : Warning<
+  "explicit initialization of field %0 may not be enforced in C++20 as type %1 "
+  "will become a non-aggregate due to the presence of user-declared "
+  "constructors">, DefaultIgnore, InGroup<CXX20Compat>;
```

Comment:
Actually looking, this is perhaps not a direct cxx20 compat warning, and should be in the explicit-init group.  What does everyone think?

Reply:
Another one I'd like Aaron's opinion here.

Reply:
Need Aaron's comment, re: should this be its own diagnostic group?

Reply:
Oof, it's a bit of both. I think the only users who would be interested in this diagnostic are users who are in older language modes and want to be compatible with newer modes, so I think that's more relevant as a compat warning. WDYT?